### ISSUE TITLE
Unable to create/use wildcard domain

### ISSUE THREAD

**Issue by justngrok** [ngrok/kubernetes-ingress-controller#408]
### Kubernetes Version

v1.27.4+k3s1

### Helm Chart Version

v3.12.3

### Ingress Controller Version

0.14.0

### System Info

MacOS M1, Sonoma 14.5

### Kubernetes Logs

```
{"level":"info","ts":"2024-08-13T15:54:00Z","logger":"cache-store-driver","msg":"sync done"}
{"level":"error","ts":"2024-08-13T15:54:00Z","logger":"controllers.ingress","msg":"Failed to sync","ingress":"default/example-ingress","error":"Domain.ingress.k8s.ngrok.com \"*-jjk8s2-ngrok-com\" is invalid: metadata.name: Invalid value: \"*-jjk8s2-ngrok-com\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')","stacktrace":"github.com/ngrok/kubernetes-ingress-controller/internal/controllers.(*IngressReconciler).Reconcile\n\tgithub.com/ngrok/kubernetes-ingress-controller/internal/controllers/ingress_controller.go:131\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:122\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:323\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:274\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:235"}
{"level":"error","ts":"2024-08-13T15:54:00Z","msg":"Reconciler error","controller":"ingress","controllerGroup":"networking.k8s.io","controllerKind":"Ingress","Ingress":{"name":"example-ingress","namespace":"default"},"namespace":"default","name":"example-ingress","reconcileID":"f9077f49-509b-48d4-ae54-c9c669273d2e","error":"Domain.ingress.k8s.ngrok.com \"*-jjk8s2-ngrok-com\" is invalid: metadata.name: Invalid value: \"*-jjk8s2-ngrok-com\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:274\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:235"}
```

### Helm Chart configuration

None

### What happened

Attempting to follow the example here:
https://ngrok.com/docs/http/?cty=k8s#wildcard-domain

As a test, reserving the wildcard domain first. Is this a limitation ingress controllers for k8s, or are we implementing the validation regex ourselves?

Have already reserved: `jjk8s2.ngrok.app`

Using `*.jjk8s2.ngrok.com` as a test wildcard (that hasn't been reserved first):

`kubectl apply ngrok-manifest-w.yaml`
"error: error parsing ngrok-manifest-w.yaml: error converting YAML to JSON: yaml: line 10: did not find expected alphabetic or numeric character"

[ngrok-manifest-w.yaml.txt](https://github.com/user-attachments/files/16602094/ngrok-manifest-w.yaml.txt)

Wrapping the "host" in quotes allows it to go through, 
`    - host: "*.jjk8s2.ngrok.com"`

`ingress.networking.k8s.io/example-ingress created`

Checking the logs, see this error:
```
{"level":"error","ts":"2024-08-13T15:54:00Z","logger":"controllers.ingress","msg":"Failed to sync","ingress":"default/example-ingress","error":"Domain.ingress.k8s.ngrok.com \"*-jjk8s2-ngrok-com\" is invalid: metadata.name: Invalid value: \"*-jjk8s2-ngrok-com\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')","stacktrace":"github.com/ngrok/kubernetes-ingress-controller/internal/controllers.
```

As a test, reserving the wildcard domain first. Is this a limitation on kubernetes, or are we implementing the regex somewhere?

Note: Same issue happens when creating using the Domains CRD.

**Comment by jonstacks** [comment]
I think this is a duplicate of [this open issue for supporting wildcard domains](https://github.com/ngrok/kubernetes-ingress-controller/issues/148).

The first issue about needing to quote the host when using a wildcard is expected as `*` is a special YAML character. We should fix the [documentation](https://ngrok.com/docs/http/?cty=k8s#wildcard-domain) for this so that it is properly quoted.

> Repeated [nodes](https://yaml.org/spec/1.2.2/#nodes) (objects) are first [identified](https://yaml.org/spec/1.2.2/#anchors-and-aliases) by an [anchor](https://yaml.org/spec/1.2.2/#anchors-and-aliases) (marked with the ampersand - “&”) and are then [aliased](https://yaml.org/spec/1.2.2/#anchors-and-aliases) (referenced with an asterisk - “*”) thereafter.


I am able to create a wildcard domain successfully using the following manifest:

```yaml
apiVersion: ingress.k8s.ngrok.com/v1alpha1
kind: Domain
metadata:
  name: my-wildcard-domain
  namespace: default
spec:
  domain: "*.mywildcarddomain.ngrok.io"
```

What I believe isn't working correctly is using wildcards via an Ingress resource like so:

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: example-ingress
  annotations:
    k8s.ngrok.com/modules: only-trusted-ips
spec:
  ingressClassName: ngrok
  rules:
    - host: "*.anotherexamplewildcard.ngrok.io"
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: blog
                port:
                  number: 80
```

I am able to recreate that and get the following error:

```
{"level":"error","ts":"2024-08-13T18:28:12Z","logger":"controllers.ingress","msg":"Failed to sync","ingress":{"name":"example-ingress","namespace":"default"},"error":"Domain.ingress.k8s.ngrok.com \"*-anotherexamplewildcard-ngrok-io\" is invalid: metadata.name: Invalid value: \"*-anotherexamplewildcard-ngrok-io\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')"}
{"level":"error","ts":"2024-08-13T18:28:12Z","msg":"Reconciler error","controller":"ingress","controllerGroup":"networking.k8s.io","controllerKind":"Ingress","Ingress":{"name":"example-ingress","namespace":"default"},"namespace":"default","name":"example-ingress","reconcileID":"e558cd33-f8ee-488b-9ee8-949aafab6b63","error":"Domain.ingress.k8s.ngrok.com \"*-anotherexamplewildcard-ngrok-io\" is invalid: metadata.name: Invalid value: \"*-anotherexamplewildcard-ngrok-io\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')"}

```

**Comment by jonstacks** [comment]
#148 should also fix this. It has been merged into `main` and will be released with 0.12.2
